File: articles/donald-knuth-and-the-art-of-computer-programming/haskell/GrowthComparison.hs
module Main where

main :: IO ()
main = do
  let n = 1000.0
  putStrLn ("log2_n=" ++ show (logBase 2 n))
  putStrLn ("n=" ++ show n)
  putStrLn ("n_log2_n=" ++ show (n * logBase 2 n))
  putStrLn ("n_squared=" ++ show (n * n))

File: articles/donald-knuth-and-the-art-of-computer-programming/rust/growth_comparison.rs
fn main() {
    let n: f64 = 1000.0;
    println!("log2_n={:.6}", n.log2());
    println!("n={:.0}", n);
    println!("n_log2_n={:.6}", n * n.log2());
    println!("n_squared={:.0}", n * n);
}
